docs/content/guides/6.multistore/3.patterns/2.design/4.different-code-per-store.md

```diff
@@ -5,4 +5,254 @@ navigation:
+
+```vue
+// /apps/stores/store-a/storefront-unified-nuxt/components/ExampleComponent/ExampleComponent.vue
+
+<template>
+  <!-- Customized className for store-a -->
+  <div class="flex items-center w-md">
+    <img :src="imageSrc" :alt="title" />
+    <h2>{{ title }}</h2>
+    <!-- Customized content for store-a -->
+  </div>
+</template>
+
+<script setup lang="ts">
+defineProps<{
+  title: string;
+  imageSrc: string;
+}>();
+</script>
+```
+
+### Outcome
+
+In both cases (Next.js and Nuxt), props remain untouched, ensuring compatibility with the rest of the application. Adjustments remain restricted to layout or presentation, avoiding unnecessary duplication or breaking changes to functionality.
+
+## Breaking Changes in Component Markup
+
+Some changes may require adjustments to the component’s **props interface**, meaning that the component is no longer compatible with its previous usage. These are referred to as **breaking changes** because they can potentially impact other parts of the application if not handled correctly.
+
+### When to Use
+
+- When introducing additional props for brand-specific requirements.
+- When removing unused or deprecated props for cleaner, leaner implementations.
+- When creating a fundamentally different component design unique to specific brands or stores.
+
+### Example Workflow for Breaking Changes
+
+#### Scenario 1: Adding Props
+
+The `ExampleComponent` requires a new prop `customSubtitle`, exclusively for `store-b`.
+
+1. **Duplicate** the component for `store-b`.
+2. **Modify the props interface** to include the new or modified prop.
+3. **Implement fallback behaviors** to ensure existing components using the old interface remain functional, or document the change clearly to propagate updates across the codebase.
+
+#### In Next.js
+
+```tsx
+// /apps/stores/store-b/storefront-unified-nextjs/components/example-component.tsx
+
+interface ExampleComponentProps {
+  title: string;
+  imageSrc: string;
+  customSubtitle?: string; // New optional prop
+}
+
+export default function ExampleComponent({ title, imageSrc, customSubtitle }: ExampleComponentProps) {
+  return (
+    <div className="store-b example-component">
+      <img src={imageSrc} alt={title} />
+      <h2>{title}</h2>
+      {customSubtitle && <p className="subtitle">{customSubtitle}</p>} {/* Customized subtitle */}
+    </div>
+  );
+}
+```
+
+#### In Nuxt
+
+```vue
+// /apps/stores/store-b/storefront-unified-nuxt/components/ExampleComponent/ExampleComponent.vue
+
+<template>
+  <div class="store-b example-component">
+    <img :src="imageSrc" :alt="title" />
+    <h2>{{ title }}</h2>
+    <!-- Customized subtitle for store-b -->
+    <p v-if="customSubtitle" class="subtitle">{{ customSubtitle }}</p>
+  </div>
+</template>
+
+<script setup lang="ts">
+defineProps<{
+   title: string;
+   imageSrc: string;
+   customSubtitle?: string; // New optional prop
+}>();
+</script>
+```
+
+#### Scenario 2: Removing Props
```

Comment:
I would also consider another approach - restructuring the parent application - removing the prop from the root store and adding it as optional to the stores that actually need it.

Reply:
Added.